tests/utils.py

```diff
@@ -2,34 +2,42 @@
 
 __all__ = ('http_server',)
 
+import socket
 from contextlib import contextmanager
 from http.server import ThreadingHTTPServer
 from pathlib import Path
 from ssl import PROTOCOL_TLS_SERVER, SSLContext
 from threading import Thread
 from typing import TYPE_CHECKING
-
-import filelock
+from urllib.parse import urlparse
 
 if TYPE_CHECKING:
     from collections.abc import Iterator
+    from http.server import HTTPServer
     from socketserver import BaseRequestHandler
     from typing import Final
 
+    from sphinx.application import Sphinx
+
 # Generated with:
 # $ openssl req -new -x509 -days 3650 -nodes -out cert.pem \
 #     -keyout cert.pem -addext "subjectAltName = DNS:localhost"
 TESTS_ROOT: Final[Path] = Path(__file__).parent
 CERT_FILE: Final[str] = str(TESTS_ROOT / 'certs' / 'cert.pem')
 
-# File lock for tests
-LOCK_PATH: Final[str] = str(TESTS_ROOT / 'test-server.lock')
-
 
 class HttpServerThread(Thread):
-    def __init__(self, handler: type[BaseRequestHandler]) -> None:
+    def __init__(self, handler: type[BaseRequestHandler], *, port: int = 0) -> None:
+        """
+        Constructs a threaded HTTP server that will bind to an arbitrary unused
+        port when it runs.
+
+        The selection of port ``0`` delegates the selection of port to Python.
```

Comment:
```suggestion
        Constructs a threaded HTTP server.  The default port number of ``0``
        delegates selection of a port number to bind to to Python.
```

Reply:
Just to be sure about this one. When you start the server thread with an address *0*, `server.server_port` is the real port number, right?

Reply:
Yep - in fact, there was a time when it didn't, but that was reported as a bug, and fixed in SVN  revision `r54253`.

Refs:
 - https://bugs.python.org/issue1531963 (bugreport)
 - https://bugs.python.org/issue1531963 (patch/fix)
 - https://github.com/python/cpython/commit/d8faa3654c2887eaa146dcdb553a9f9793bd2e5a (`git` history commit containing the SVN revision merge)

And to confirm that:

```pyi
>>> import socketserver
>>> s = socketserver.TCPServer(("0.0.0.0", 0), None)
>>> s.server_address
('0.0.0.0', 60527)
```